src/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -352,9 +360,11 @@ static smt_termt convert_expr_to_smt(const ieee_float_op_exprt &float_operation)
 
 static smt_termt convert_expr_to_smt(const mod_exprt &truncation_modulo)
 {
-  UNIMPLEMENTED_FEATURE(
-    "Generation of SMT formula for truncation modulo expression: " +
-    truncation_modulo.pretty());
```

Comment:
:pick: We expect this operator to be applied only to (signed/unsigned) integer bit vectors. It might be good to document this expectation using a `UNEXPECTEDCASE` macro, similar to how `smt2_convt::convert_mod` does this.